crates/rolldown_plugin_replace/src/plugin.rs

```diff
@@ -38,8 +33,25 @@ pub struct ReplacePlugin {
   sourcemap: bool,
 }
 
-static NON_ASSIGNMENT_MATCHER: LazyLock<Regex> =
-  LazyLock::new(|| Regex::new("\\b(?:const|let|var)\\s+$").expect("Should be valid regex"));
+// Checks if the given string ends with a variable declaration keyword (const, let, var)
+// followed by whitespace, which would indicate the start of a variable declaration.
+fn is_variable_declaration_prefix(s: &str) -> bool {
+  // First check if there's any whitespace at the end
+  if !s.ends_with(|c: char| c.is_whitespace()) {
+    return false;
+  }
+
+  // Trim the trailing whitespace
+  let s = s.trim_end();
+
+  // Check for word boundary before the keywords
+  (s.ends_with("const")
+    && (s.len() == 5 || !s.chars().nth(s.len() - 6).unwrap_or(' ').is_alphanumeric()))
```

Comment:
This seems  inefficient, as chars().nth() has a time complexity of O(N).

Reply:
```
running 2 tests
test tests::pure_function ... bench:          80.56 ns/iter (+/- 12.09)
test tests::regex         ... bench:         201.07 ns/iter (+/- 6.74)
``` 
also updated the pr description